AWSAuthSDK/Sources/AWSMobileClient/AWSMobileClientExtensions.swift

```diff
@@ -288,36 +279,47 @@ extension AWSMobileClient {
                     self.customRoleArnInternal = customRoleArn
                     _AWSMobileClient.sharedInstance().setCustomRoleArnInternal(customRoleArn, for: self)
                 }
-                self.performFederatedSignInTasks(provider: providerName, token: token)
-                // If any credentials operation is pending, we invoke the waiting block to resume with new credentials
-                if (self.pendingAWSCredentialsCompletion != nil) {
-                    self.internalCredentialsProvider?.credentials().continueWith(block: { (task) -> Any? in
-                        if let credentials = task.result {
-                            self.pendingAWSCredentialsCompletion?(credentials, nil)
-                        } else if let error = task.error {
-                            self.pendingAWSCredentialsCompletion?(nil, error)
-                        }
+                self.performFederatedSignInTasks(provider: providerName, token: token).continueWith(block: { (task) -> Any? in
+                    // If any credentials operation is pending, we invoke the waiting block to resume with new credentials
+                    if let credentials = task.result {
```

Comment:
I think this is correct, but to be sure: Do we need to inspect the credentials here, or is the existence of them sufficient to assert that the state is `signedIn`?

Reply:
I guess just checking the taskresult is good for now so that I can just concentrate on moving the race condition out. Will take this as a follow up if we need to check for more values in another PR.